RecoJets/JetProducers/plugins/CSJetProducer.cc

```diff
@@ -78,30 +99,40 @@ void CSJetProducer::runAlgorithm(edm::Event& iEvent, edm::EventSetup const& iSet
     if (nParticles == 0)
       continue;  //don't subtract ghost jets
 
-    //assign rho and rhom to ghosts according to local eta-dependent map
-    double rho = 1e-6;
-    double rhom = 1e-6;
+    //assign rho and rhom to ghosts according to local eta-dependent map + modulation as function of phi
     for (fastjet::PseudoJet& ighost : ghosts) {
+      double rhoModulationFactor = 1.;
+      double ghostPhi = ighost.phi_std();
+
+      if (useModulatedRho_) {
+        if (!rhoFlowFitParams->empty()) {
+          if (minProb && maxProb) {
+            rhoModulationFactor = getModulatedRhoFactor(ghostPhi,
+                                                        rhoFlowFitParams->at(2),
+                                                        rhoFlowFitParams->at(4),
+                                                        rhoFlowFitParams->at(1),
+                                                        rhoFlowFitParams->at(3));
+          }
+        }
+      }
+
+      int32_t ghostPos = -1;
       if (ighost.eta() <= etaRanges->at(0) || bkgVecSize == 1) {
```

Comment:
```suggestion
      auto const ighostEta = ighost.eta();
      if (ighostEta <= etaRanges->at(0) || bkgVecSize == 1) {
```
apparently I'm (re)discovering for myself that the compiler wouldn't necessarily optimize away repeated calls

This minor optimization of fastjet::PseudoJet::pseudorapidity() calls will save about 3% of RECO time (ignoring startup and IO cost) in wf 158.0 
- before https://slava77sk.web.cern.ch/slava77sk/reco/cgi-bin/igprof-navigator/CMSSW_11_2_X_2020-07-18-1100-test30057.158.0.step3.prod.100.pp/102
- after https://slava77sk.web.cern.ch/slava77sk/reco/cgi-bin/igprof-navigator/CMSSW_11_2_X_2020-07-18-1100-test30057.19f86335b13.158.0.step3.prod.100.pp/164

Reply:
> This minor optimization

the ```s/ighost.eta()/ighostEta/```
to apply to the full scope here